### PR TITLE
Upgrade to react 16

### PR DESCRIPTION
**Pull request by PeterSiegl**

### CODE REVIEW

File: src/components/Accordion/__tests__/Accordion.spec.js [hunk 1]
```diff
@@ -175,7 +175,7 @@ describe('<Accordion />', () => {
   });
 
   it('takes styled prop and wraps Accordion in slds-card', () => {
-    const mounted = render(
+    const mounted = mount(
```
Comment (fspoettel) [inline src/components/Accordion/__tests__/Accordion.spec.js]: Why exactly did we render here in the first place? 🙈

File: src/components/Lookup/__tests__/Lookup.spec.js [hunk 2]
```diff
@@ -144,10 +144,9 @@ describe('<Lookup />', () => {
 
   it('renders selections correctly', () => {
     mounted.setState({ selected: [sampleData[0]] });
-
     const firstData = sampleData[0];
     const result = mounted.find('.slds-pill');
-    expect(result.find('.slds-pill__remove').length).toBe(1);
+    expect(result.find('button.slds-pill__remove').length).toBe(1);
```
Comment (fspoettel) [inline src/components/Lookup/__tests__/Lookup.spec.js]: Why were these failing?  `expect(result.find('.slds-pill__remove').length).toBe(1)` should not have failed by the looks of it.

File: src/components/Form/__tests__/Checkbox.spec.js [hunk 3]
```diff
@@ -18,8 +18,8 @@ describe('<Checkbox />', () => {
     mounted = mount(<Checkbox {...props} />);
   });
 
-  it('renders the id', () => {
-    expect(mounted.find(`#${props.id}`).length).toEqual(1);
+  it('renders the component', () => {
+    expect(mounted.find(Checkbox).length).toEqual(1);
```
Comment (fspoettel) [inline src/components/Form/__tests__/Checkbox.spec.js]: Why does this not cover the original spec anymore?

File: src/components/Form/__tests__/Input.spec.js [hunk 4]
```diff
@@ -24,8 +24,8 @@ describe('<Input />', () => {
     mounted = mount(<Input {...props} />);
   });
 
-  it('renders the id', () => {
-    expect(mounted.find(`#${props.id}`).length).toEqual(1);
+  it('renders the component', () => {
+    expect(mounted.find(Input).length).toEqual(1);
```
Comment (fspoettel) [inline src/components/Form/__tests__/Input.spec.js]: Same as above

File: src/components/Accordion/__tests__/Accordion.spec.js (hunk 1 above)
Comment (PeterSiegl) [reply src/components/Accordion/__tests__/Accordion.spec.js]: I dunno 😄

File: src/components/Lookup/__tests__/Lookup.spec.js (hunk 2 above)
Comment (PeterSiegl) [reply src/components/Lookup/__tests__/Lookup.spec.js]: there are now several elements that have the `slds-pill__remove` class: `Button`, `ButtonIcon` and `button`

File: src/components/Form/__tests__/Checkbox.spec.js (hunk 3 above)
Comment (PeterSiegl) [reply src/components/Form/__tests__/Checkbox.spec.js]: will add the check for the id again 👍

File: src/components/Accordion/__tests__/Accordion.spec.js (hunk 1 above)
Comment (brunottonurb) [reply src/components/Accordion/__tests__/Accordion.spec.js]: It was my first Component :smiley:

File: internals/shim.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+global.requestAnimationFrame = (callback) => {
```
[the remark this answers is not in the document]
Comment (fspoettel) [reply internals/shim.js]: @PeterSiegl 👆

[the remark this answers is not in the document]
Comment (PeterSiegl) [reply internals/shim.js]: image
we need this to fix the above warnings when testing.

**Pull request merged**